drivers/temperature/adt7420/adt7420.c

```diff
@@ -40,55 +40,107 @@
 /******************************************************************************/
 /***************************** Include Files **********************************/
 /******************************************************************************/
+#include <stdint.h>
 #include <stdlib.h>
+#include "spi.h"
+#include "i2c.h"
+#include "error.h"
 #include "adt7420.h"
 
+const struct adt7420_chip_info chip_info[] = {
+	[ID_ADT7410] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7420] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7422] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7310] = {
+		.resolution = 16,
+		.communication = SPI,
+	},
+	[ID_ADT7320] = {
+		.resolution = 16,
+		.communication = SPI,
+	}
+};
+
 /***************************************************************************//**
- * @brief Reads the value of a register.
+ * @brief Reads the value of a register SPI/I2C interface.
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
  *
  * @return register_value  - Value of the register.
 *******************************************************************************/
-uint8_t adt7420_get_register_value(struct adt7420_dev *dev,
-				   uint8_t register_address)
+uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+				    uint8_t register_address)
 {
 	uint8_t register_value = 0;
+	uint8_t data[2] = {0,0xFF};
+
+	if (chip_info[dev->active_device].communication == SPI) {
+		/* Form the SPI register read command byte. Bits [5:3] of command byte
+		 holds the register address.*/
+		data[0] = ADT7320_READ_CMD  | (register_address << 3);
+
+		return set_shift_reg(dev, register_address, data);
+	} else {
+		data[0] = register_address;
 
-	i2c_write(dev->i2c_desc,
-		  &register_address,
-		  1,
-		  0);
-	i2c_read(dev->i2c_desc,
-		 &register_value,
-		 1,
-		 1);
+		i2c_write(dev->i2c_desc,
+			  &register_address,
+			  1,
+			  0); //add a repeat start
+		i2c_read(dev->i2c_desc,
+			 &register_value,
+			 1,
+			 1);
 
-	return register_value;
+		return register_value;
+	}
 }
 
 /***************************************************************************//**
- * @brief Sets the value of a register.
+ * @brief Sets the value of a register SPI/I2C.
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
- * @param register_value   - Value of the register.
+ * @param num_data_bytes   - Number of bytes.
+ * @param data             - Data to be written in input register.
  *
  * @return None.
 *******************************************************************************/
-void adt7420_set_register_value(struct adt7420_dev *dev,
-				uint8_t register_address,
-				uint8_t register_value)
+void set_register_value(struct adt7420_dev *dev,
```

Comment:
should return a return code

Reply:
The function has been re-framed in such a way that it returns an int32_t value. Modified function prototype for the same as well.